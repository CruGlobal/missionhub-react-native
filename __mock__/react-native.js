import { NativeModules } from 'react-native';

// From https://github.com/facebook/react-native/issues/18279#issuecomment-374177940
// Will be made unnecessary once https://github.com/facebook/react-native/pull/18718 is released. It doesn't seem to be in 0.55.3
NativeModules.BlobModule = {
  ...NativeModules.BlobModule,
  addNetworkingHandler: jest.fn(),
};
