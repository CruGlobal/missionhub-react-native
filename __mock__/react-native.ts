jest.mock(
  'react-native/Libraries/Utilities/NativePlatformConstantsIOS',
  () => ({
    // @ts-ignore
    ...jest.requireActual(
      'react-native/Libraries/Utilities/NativePlatformConstantsIOS',
    ),
    getConstants: () => ({
      forceTouchAvailable: false,
      interfaceIdiom: 'en',
      isTesting: false,
      osVersion: 'ios',
      reactNativeVersion: { major: 60, minor: 1, patch: 0 },
      systemName: 'ios',
    }),
  }),
);
