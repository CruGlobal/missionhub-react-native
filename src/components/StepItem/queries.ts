import gql from 'graphql-tag';

import { REMINDER_BUTTON_FRAGMENT } from '../ReminderButton/queries';
import { REMINDER_DATE_TEXT_FRAGMENT } from '../ReminderDateText/queries';

export const STEP_ITEM_FRAGMENT = gql`
  fragment StepItem on Step {
    id
    title
    completedAt
    receiver {
      id
      fullName
    }
    community {
      id
    }
    reminder {
      ...ReminderButton
      ...ReminderDateText
    }
  }
  ${REMINDER_BUTTON_FRAGMENT}
  ${REMINDER_DATE_TEXT_FRAGMENT}
`;
