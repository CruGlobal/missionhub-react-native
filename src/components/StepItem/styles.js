
import { StyleSheet } from 'react-native';
import theme from '../../theme';

export default StyleSheet.create({
  row: {
    height: theme.itemHeight,
    paddingHorizontal: 24,
    backgroundColor: theme.convert({ color: theme.white, alpha: 0 }),
    width: theme.fullWidth,
  },
  swipeable: {
    marginTop: 1,
    backgroundColor: theme.convert({ color: theme.secondaryColor, lighten: 0.4 }),
  },
  listSwipeable: {
    marginTop: 1,
    backgroundColor: theme.white,
    borderBottomWidth: theme.separatorHeight,
    borderBottomColor: theme.separatorColor,
    height: 70,
  },
  draggable: {
    borderBottomWidth: theme.separatorHeight,
    borderBottomColor: theme.separatorColor,
  },
  dragging: {
    backgroundColor: theme.convert({ color: theme.white, alpha: 0.8 }),
    shadowColor: 'black',
    shadowOffset: { width: 0, height: 0 },
    shadowOpacity: 0.8,
    shadowRadius: 7,
    elevation: 8,
  },
  offscreen: {
    opacity: 0,
  },
  person: {
    fontSize: 16,
    fontWeight: 'bold',
    color: theme.primaryColor,
  },
  description: {
    fontSize: 14,
  },
});
