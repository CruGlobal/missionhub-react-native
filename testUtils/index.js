import 'react-native';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Need to import for all translation screens to work properly
import '../src/i18n';

Enzyme.configure({ adapter: new Adapter() });

export const createMockStore = (state = {}) => {
  return {
    getState: jest.fn(() => (state)),
    dispatch: jest.fn((response) => Promise.resolve(response)),
    subscribe: jest.fn(),
  };
};

export const createMockNavState = (params = {}) => {
  return { state: { params } };
};

export const testSnapshot = (data) => {
  expect(renderer.create(data)).toMatchSnapshot();
};

export const renderShallow = (component, store) => {
  let renderedComponent = shallow(
    component,
    { context: { store: store } }
  );

  // If component has translation wrappers, dive deeper
  while (renderedComponent.is('Translate') || renderedComponent.is('I18n')) {
    renderedComponent = renderedComponent.dive();
  }

  // Render contents of component
  renderedComponent = renderedComponent.dive();
  return renderedComponent;
};

export const testSnapshotShallow = (component, store) => {
  const renderedComponent = renderShallow(component, store);
  expect(renderedComponent).toMatchSnapshot();
  return renderedComponent;
};

export const mockFnWithParams = (obj, method, expectedReturn, ...expectedParams) => {
  return obj[method] = jest.fn(
    (...actualParams) => {
      expect(actualParams).toEqual(expectedParams);
      return expectedReturn;
    }
  );
};
