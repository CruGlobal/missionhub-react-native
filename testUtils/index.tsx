import React, { ReactElement, ReactNode } from 'react';
import 'react-native';
import { Provider } from 'react-redux';
import { Provider as ProviderLegacy } from 'react-redux-legacy';
import thunk from 'redux-thunk';
import configureStore, { MockStore } from 'redux-mock-store';
// eslint-disable-next-line import/named
import { NavigationParams, NavigationProvider } from 'react-navigation';
import { ApolloProvider } from '@apollo/react-hooks';
import { ReactTestRendererJSON } from 'react-test-renderer';
import { render, RenderAPI } from 'react-native-testing-library';
import snapshotDiff from 'snapshot-diff';
import Enzyme, { shallow as enzymeShallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { IMocks } from 'graphql-tools';

import { createApolloMockClient } from './apolloMockClient';
import { createNavigationProp } from './navigationHelpers';

Enzyme.configure({ adapter: new Adapter() });

export const createThunkStore = configureStore([thunk]);

interface RenderWithContextParams {
  initialState?: {};
  store?: MockStore;
  navParams?: NavigationParams;
  mocks?: IMocks;
  noWrappers?: boolean;
}

// Inspiration from https://github.com/kentcdodds/react-testing-library/blob/52575005579307bcfbe7fbe4ef4636147c03c6fb/examples/__tests__/react-redux.js#L69-L80
export function renderWithContext(
  component: ReactElement,
  {
    initialState,
    store = createThunkStore(initialState),
    navParams,
    mocks: mocks = {},
    noWrappers = false,
  }: RenderWithContextParams = {},
) {
  const mockApolloClient = createApolloMockClient(mocks);

  const navigation = createNavigationProp(navParams);

  let renderResult: RenderAPI;
  if (noWrappers) {
    renderResult = render(React.cloneElement(component, { navigation }));
  } else {
    // Warning: don't call any functions in here that return new instances on every call. All the props need to stay the same otherwise rerender won't work.
    const wrapper = ({ children }: { children: ReactNode }) => (
      <NavigationProvider value={navigation}>
        <ProviderLegacy store={store}>
          <Provider store={store}>
            <ApolloProvider client={mockApolloClient}>
              {children}
            </ApolloProvider>
          </Provider>
        </ProviderLegacy>
      </NavigationProvider>
    );

    renderResult = render(React.cloneElement(component, { navigation }), {
      wrapper,
    });
  }

  let storedSnapshot: ReactTestRendererJSON | null;
  return {
    ...renderResult,
    store,
    snapshot: () => expect(renderResult.toJSON()).toMatchSnapshot(),
    recordSnapshot: () => (storedSnapshot = renderResult.toJSON()),
    diffSnapshot: () => {
      if (!storedSnapshot) {
        throw new Error(
          'You must call recordSnapshot to store an initial snapshot before calling diffSnapshot',
        );
      }
      expect(
        snapshotDiff(storedSnapshot, renderResult.toJSON()),
      ).toMatchSnapshot();
    },
  };
}

// TODO: Remove all legacy rendering functions below

export const createMockNavState = (params = {}) => {
  return { state: { params } };
};

export const testSnapshot = (component: ReactElement) => {
  const { toJSON } = render(component);
  expect(toJSON()).toMatchSnapshot();
};

export const renderShallow = (
  component: ReactElement,
  store = createThunkStore(),
) => {
  let renderedComponent = enzymeShallow(
    <ProviderLegacy store={store}>{component}</ProviderLegacy>,
  ).dive();

  // If component has translation wrappers, dive deeper
  while (
    // eslint-disable-next-line  @typescript-eslint/no-explicit-any
    ((renderedComponent.type() as any).displayName || '').startsWith(
      'withI18nextTranslation(',
    )
  ) {
    renderedComponent = renderedComponent.dive();
  }

  // Render contents of component
  renderedComponent = renderedComponent.dive();
  return renderedComponent;
};

export const testSnapshotShallow = (
  component: ReactElement,
  store = createThunkStore(),
) => {
  const renderedComponent = renderShallow(component, store);
  expect(renderedComponent).toMatchSnapshot();
  return renderedComponent;
};
