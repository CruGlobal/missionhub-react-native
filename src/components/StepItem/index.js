import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Flex, Text } from '../common';
import styles from './styles';

export default class StepItem extends Component {
  setNativeProps(nProps) { this._view.setNativeProps(nProps); }
  render() {
    const { step, type, isMe } = this.props;
    let ownerName = isMe ? 'Me' : step.receiver.full_name || '';
    ownerName = ownerName.toUpperCase();
    return (
      <Flex
        ref={(c) => this._view = c}
        justify="center"
        style={[
          styles.row,
          type && styles[type] ? styles[type] : null,
        ]}
      >
        {
          type === 'listSwipeable' ? null : (
            <Text style={styles.person}>
              {ownerName}
            </Text>
          )
        }
        <Text style={styles.description}>
          {step.title}
        </Text>
      </Flex>
    );
  }

}

StepItem.propTypes = {
  step: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    accepted_at: PropTypes.date,
    completed_at: PropTypes.date,
    created_at: PropTypes.date,
    updated_at: PropTypes.date,
    notified_at: PropTypes.date,
    note: PropTypes.string,
    owner: PropTypes.object.isRequired,
    receiver: PropTypes.object.isRequired,
  }).isRequired,
  isMe: PropTypes.bool,
  type: PropTypes.oneOf(['draggable', 'swipeable', 'listSwipeable', 'dragging', 'offscreen']),
};
