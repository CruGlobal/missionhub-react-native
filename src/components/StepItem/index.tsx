import React from 'react';
import { View, Image, Text } from 'react-native';
import { useTranslation } from 'react-i18next';
import { useMutation } from '@apollo/react-hooks';
import { useDispatch } from 'react-redux';

import { Icon, Card, Touchable, Button } from '../common';
import ItemHeaderText from '../ItemHeaderText';
import ReminderButton from '../ReminderButton';
import ReminderDateText from '../ReminderDateText';
import { useIsMe } from '../../utils/hooks/useIsMe';
import GREY_CHECKBOX from '../../../assets/images/checkIcon-grey.png';
import BLUE_CHECKBOX from '../../../assets/images/checkIcon-blue.png';
import { navigatePush } from '../../actions/navigation';
import { ACCEPTED_STEP_DETAIL_SCREEN } from '../../containers/AcceptedStepDetailScreen';
import { COMPLETED_STEP_DETAIL_SCREEN } from '../../containers/CompletedStepDetailScreen';
import { navToPersonScreen } from '../../actions/person';
import { handleAfterCompleteStep } from '../../actions/steps';
import { COMPLETE_STEP_MUTATION } from '../../containers/AcceptedStepDetailScreen/queries';
import { CONTACT_STEPS } from '../../constants';
import { StepTypeBadge } from '../StepTypeBadge/StepTypeBadge';
import {
  CompleteStep,
  CompleteStepVariables,
} from '../../containers/AcceptedStepDetailScreen/__generated__/CompleteStep';

import { StepItem as Step } from './__generated__/StepItem';
import styles from './styles';

interface StepItemProps {
  step: Step;
  showName?: boolean;
  showCheckbox?: boolean;
  testID?: string;
}

const StepItem = ({
  step,
  showName = true,
  showCheckbox = true,
}: StepItemProps) => {
  const { t } = useTranslation();
  const dispatch = useDispatch();

  const [completeStep] = useMutation<CompleteStep, CompleteStepVariables>(
    COMPLETE_STEP_MUTATION,
    {
      onCompleted: data => {
        data.markStepAsCompleted?.step &&
          dispatch(
            handleAfterCompleteStep(
              {
                id: data.markStepAsCompleted?.step?.id,
                receiver: data.markStepAsCompleted?.step?.receiver,
                community: data.markStepAsCompleted?.step?.community,
              },
              CONTACT_STEPS,
            ),
          );
      },
    },
  );

  const onPressCard = () => {
    dispatch(
      navigatePush(
        step.completedAt
          ? COMPLETED_STEP_DETAIL_SCREEN
          : ACCEPTED_STEP_DETAIL_SCREEN,
        { stepId: step.id, personId: step.receiver.id },
      ),
    );
  };

  const onPressName = () => {
    const { receiver } = step;
    dispatch(navToPersonScreen(receiver.id));
  };

  const onPressCheckbox = async () => {
    step && (await completeStep({ variables: { input: { id: step.id } } }));
  };

  const isMe = useIsMe(step.receiver && step.receiver.id);
  const ownerName = isMe
    ? t('me')
    : (step.receiver && step.receiver.fullName) || '';
  const {
    cardHeader,
    bellIcon,
    reminderButton,
    iconButton,
    checkIcon,
  } = styles;
  return (
    <Card testID="StepItemCard" onPress={onPressCard} style={styles.card}>
      <View style={{ flex: 1 }}>
        <View style={cardHeader}>
          {showName ? (
            <Touchable
              testID="StepItemPersonButton"
              onPress={onPressName}
              style={styles.nameWrap}
            >
              <ItemHeaderText style={styles.stepUserName} text={ownerName} />
            </Touchable>
          ) : (
            <StepTypeBadge stepType={step.stepType} />
          )}
        </View>
        <Text style={styles.description}>{step.title}</Text>
        {step.completedAt ? null : (
          <View style={{ alignSelf: 'flex-start' }}>
            <ReminderButton
              testID="StepReminderButton"
              stepId={step.id}
              reminder={step.reminder}
            >
              <View style={reminderButton}>
                {step.reminder ? (
                  <Icon name="bellIcon" type="MissionHub" style={bellIcon} />
                ) : null}
                <ReminderDateText reminder={step.reminder} />
              </View>
            </ReminderButton>
          </View>
        )}
      </View>
      {showCheckbox ? (
        <Button
          testID="CompleteStepButton"
          onPress={onPressCheckbox}
          style={iconButton}
          disabled={!!step.completedAt}
        >
          <Image
            source={step.completedAt ? GREY_CHECKBOX : BLUE_CHECKBOX}
            style={checkIcon}
          />
        </Button>
      ) : null}
    </Card>
  );
};

export default StepItem;
