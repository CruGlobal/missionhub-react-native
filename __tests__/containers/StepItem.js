import 'react-native';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { createMockStore, renderShallow, testSnapshotShallow } from '../../testUtils';
import StepItem from '../../src/components/StepItem';

const date = '2017-12-06T14:24:52Z';
const mockStep = {
  id: '1',
  title: 'Test Step',
  accepted_at: date,
  completed_at: date,
  created_at: date,
  updated_at: date,
  notified_at: date,
  note: 'Note',
  owner: { id: '456' },
  receiver: { id: '456' },
};


const mockState = {
  auth: {
    person: {
      id: '1',
    },
  },
};

const store = createMockStore(mockState);

jest.mock('react-native-device-info');


it('renders me correctly', () => {
  testSnapshotShallow(
    <StepItem step={mockStep} />,
    store,
  );
});

it('renders not me correctly', () => {
  testSnapshotShallow(
    <StepItem step={{
      ...mockStep,
      id: '2',
    }} />,
    store,
  );
});

it('renders type swipeable correctly', () => {
  testSnapshotShallow(
    <StepItem step={mockStep} type="swipeable" />,
    store,
  );
});

it('renders type contact correctly', () => {
  testSnapshotShallow(
    <StepItem step={mockStep} type="contact" />,
    store,
  );
});

it('renders type reminder correctly', () => {
  testSnapshotShallow(
    <StepItem step={mockStep} type="reminder" />,
    store,
  );
});

it('renders type action correctly', () => {
  testSnapshotShallow(
    <StepItem step={mockStep} type="swipeable" onAction={() => {}} />,
    store,
  );
});

it('renders hover for step', () => {
  const component = renderShallow(
    <StepItem step={mockStep} type="swipeable" onAction={() => { }} />,
    store
  );
  component.setState({ hovering: true });
  expect(component).toMatchSnapshot();
});

describe('step item methods', () => {
  let component;
  const mockSelect = jest.fn();
  beforeEach(() => {
    Enzyme.configure({ adapter: new Adapter() });
    const screen = shallow(
      <StepItem step={{ ...mockStep, receiver: null }} onSelect={mockSelect} type="swipeable" onAction={() => { }} />,
      { context: { store } },
    );

    component = screen.dive().dive().dive().instance();
  });

  it('handles create interaction', () => {
    component.handleSelect();
    expect(mockSelect).toHaveBeenCalledTimes(0);
  });


});
