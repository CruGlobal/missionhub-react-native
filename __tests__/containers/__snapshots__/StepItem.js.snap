// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders me correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits="button"
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "marginTop": 1,
          "paddingLeft": 24,
          "paddingRight": 13,
          "paddingVertical": 16,
          "width": 750,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "color": "#007398",
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        Test Step
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders not me correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits="button"
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "marginTop": 1,
          "paddingLeft": 24,
          "paddingRight": 13,
          "paddingVertical": 16,
          "width": 750,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "color": "#007398",
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        Test Step
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders type action correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits="button"
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "backgroundColor": "#ffffff",
        "borderBottomColor": "#E6E8EC",
        "borderBottomWidth": 1,
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "marginTop": 1,
          "paddingLeft": 24,
          "paddingRight": 13,
          "paddingVertical": 16,
          "width": 750,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "color": "#007398",
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        Test Step
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits="button"
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#ffffff",
              },
              Object {
                "fontSize": 18,
              },
              Array [
                Object {
                  "color": "#52C5DC",
                  "fontSize": 35,
                  "padding": 10,
                },
                undefined,
              ],
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        Ó§è
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders type contact correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits="button"
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "backgroundColor": "#ffffff",
        "borderBottomColor": "#E6E8EC",
        "borderBottomWidth": 1,
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "marginTop": 1,
          "paddingLeft": 24,
          "paddingRight": 13,
          "paddingVertical": 16,
          "width": 750,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        Test Step
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders type reminder correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits="button"
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(183, 231, 240)",
      "borderBottomColor": "#E6E8EC",
      "borderBottomWidth": 1,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "marginTop": 1,
          "paddingLeft": 24,
          "paddingRight": 13,
          "paddingVertical": 16,
          "width": 750,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "color": "#007398",
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        Test Step
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders type swipeable correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits="button"
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "backgroundColor": "#ffffff",
        "borderBottomColor": "#E6E8EC",
        "borderBottomWidth": 1,
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "marginTop": 1,
          "paddingLeft": 24,
          "paddingRight": 13,
          "paddingVertical": 16,
          "width": 750,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "color": "#007398",
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        Test Step
      </Text>
    </View>
  </View>
</View>
`;
